bank_component.rb

```diff
@@ -0,0 +1,97 @@
+=begin
+ Feature 1: 
+ This feature will allow user to either sign up for an account.
+ Or log in with their last name,last 4 digits of their social security number, and password.        
+=end
+
+
+
+#class for log in and sign up method
+class LogInSignUp
+
+    @@user_info=[
+        {first:'andy', last:'park', password:'hello123', ssn:'123456789'},
+        {first:'Jason', last:'kim', password:'bye123', ssn:'135791098'}
+    ]
+
+    #Instance variable to determine if the user's credential is validated
+    @user_credential_validation = false
+    
+    attr_accessor :last_name, :password, :ssn
+
+    def initialize(last_name, password, ssn)
+        @last_name = last_name
+        @password=password
+        @ssn=ssn
+    end
+
+    def validate_log_in_credentials
+        @@user_info.each do |x|
+            if x[:last].to_s == @last_name && x[:password] == @password && x[:ssn][5,8] == @ssn
+                puts "Welcome back #{x[:first]} #{x[:last]}!"
+                return
+            else
+                puts "sorry access denied!"
+                return
```

Comment:
This `else` block might have unintentional side effects. i.e. There might be a bug here.

I believe this `return` will cause the whole method to return if the first user does not match. Meaning, the second user will never be checked.

To fix this, we'll probably need to avoid these returns altogether. Though it would be fine to still return for the match case.

For the non-match case, we might need to remove the `else` completely, then output `"sorry access denied!"` after the `each` loop finishes. This way we will only make it past the each loop if all users failed to match the credentials. If there was a match, the early return would have skipped the message after the loop.

Another approach would be to use local vars to track the state of things.